### PR TITLE
Batch publishing

### PR DESCRIPTION
**Pull request by joskfg**
Initial implementation.

I didn't found an easy way to do this due to the object is created in the final publication, so I need a store. It would be google to inject the dependency and be able to do a pass to the real methods instead of accumulate in the object and do all the work in the final publishing.

Any suggestion?

### CODE REVIEW

File: src/Amqp.php [hunk 1]
```diff
@@ -34,6 +36,51 @@ public function publish($routing, $message, array $properties = [])
...
+
+            $publisher->batchBasicPublish($messageData['routing'], $messageData['message']);
+        }
+
+        $publisher->batchPublish();
+        $this->forgetBatchedMessages();
+        Request::shutdown($publisher->getChannel(), $publisher->getConnection());
+    }
+
+    public function forgetBatchedMessages()
```
Comment (stevenklar) [inline src/Amqp.php:79]: This shoudl be private and docblocked.

**Review by stevenklar — CHANGES_REQUESTED** [review]
Please fix the docblocks and function scope as mentioned.

File: src/Publisher.php [hunk 2]
```diff
@@ -17,4 +17,18 @@ public function publish($routing, $message)
...
+    /**
+     * @param string $routing
+     * @param Message $message
+     */
+    public function batchBasicPublish($routing, $message)
+    {
+        $this->getChannel()->batch_basic_publish($message, $this->getProperty('exchange'), $routing);
+    }
+
+    public function batchPublish()
```
Comment (stevenklar) [inline src/Publisher.php:30]: Can you please add a proper docblock here?

**Comment by joskfg** [comment]
If you consider OK the approach, I'll fix the code.

**Review by stevenklar — APPROVED** [review]

**Comment by stevenklar** [comment]
Thank you for the contribution.
The method names might be open to change until next release, but I will merge it for now.

**Pull request merged**